cmd/kubeadm/app/util/pkiutil/pki_helpers.go

```diff
@@ -656,9 +657,10 @@ func NewSignedCert(cfg *CertConfig, key crypto.Signer, caCert *x509.Certificate,
 			CommonName:   cfg.CommonName,
 			Organization: cfg.Organization,
 		},
-		DNSNames:              cfg.AltNames.DNSNames,
-		IPAddresses:           cfg.AltNames.IPs,
-		SerialNumber:          serial,
+		DNSNames:    cfg.AltNames.DNSNames,
+		IPAddresses: cfg.AltNames.IPs,
+		// add 1 to serial to make it a uniform random value in [1, max).
+		SerialNumber:          new(big.Int).Add(serial, big.NewInt(1)),
```

Comment:
my suggestion would be to prepare `serial` with the  +1 in advance and just have the `SerialNumber: serial` in cert structs. also write a single comment explaining briefly the range `[1, max-1]` generation.

applies to other locations too.